Sming/Components/MultipartParser/MultipartParser.cpp

```diff
@@ -62,83 +62,113 @@ size_t formMultipartParser(HttpRequest& request, const char* at, int length)
 		return -1;                                                                                                     \
 	}
 
-MultipartParser::MultipartParser(HttpRequest* request)
-{
-	if(request->headers.contains(HTTP_HEADER_CONTENT_TYPE)) {
-		// Content-Type: multipart/form-data; boundary=------------------------a48863c0572edce6
-		int startPost = request->headers[HTTP_HEADER_CONTENT_TYPE].indexOf("boundary=");
-		if(startPost == -1) {
-			return;
-		}
-
-		startPost += 9;
-		String boundary = "--" + request->headers[HTTP_HEADER_CONTENT_TYPE].substring(startPost);
-		parser = multipart_parser_init(boundary.c_str(), &settings);
-		parser->data = this;
-	}
 
-	this->request = request;
+MultipartParser::MultipartParser(HttpRequest& request, const String& boundary) 
+	: request(request), boundary(boundary)
+{
+	// Note: Storing the request by reference makes this object noncopyable and as a result the underlying memory of the boundary string does not change throughout the lifetime of this object.
+	multipart_parser_init(&parserEngine, boundary.c_str(), boundary.length(), &settings);
+	parserEngine.data = this;
 }
 
-MultipartParser::~MultipartParser()
+MultipartParser *MultipartParser::create(HttpRequest& request)
 {
-	multipart_parser_free(parser);
-	parser = nullptr;
+	if(request.headers.contains(HTTP_HEADER_CONTENT_TYPE)) {
+		// Content-Type: multipart/form-data; boundary=------------------------a48863c0572edce6
+		int startPost = request.headers[HTTP_HEADER_CONTENT_TYPE].indexOf(_F("boundary="));
```

Comment:
Optimisation: `indexOf(_F("boundary="), 9)`

Reply:
Is there any downside in using `indexOf(FS("boundary="))` instead?
At least for small strings and thanks to your SSO, `WString` should not do a heap allocation, so both `_F` and `FS` with its implicit `operator WString` will essentially do a `memcpy_aligned`.

Reply:
Good point. With SSO `indexOf(F("boundary="))` is also an option.